### DISCUSSION TITLE
Upstream Software Releases & End of EOL grace period

### DISCUSSION THREAD

**Discussion by xnox** [wolfi-dev/os#27195] [General]
New releases:

* January
  - [Java Quarterly Patch Update](https://www.oracle.com/uk/security-alerts/)
* February
  - [Glibc](https://sourceware.org/glibc/wiki/Glibc%20Timeline)
  - [Go](https://go.dev/doc/devel/release)
* March
  - [Java](https://www.oracle.com/java/technologies/java-se-support-roadmap.html)
  - [LLVM](https://releases.llvm.org/)
* April
  - [OpenSSL](https://openssl-library.org/post/2023-08-29-steps-forward/)
  - [NodeJS](https://github.com/nodejs/Release/blob/main/schedule.json)
  - [Java Quarterly Patch Update](https://www.oracle.com/uk/security-alerts/)
  - [Mozilla CA removals](https://wiki.mozilla.org/CA/Root_CA_Lifecycles)
* May
  - [GCC](https://gcc.gnu.org/develop.html#timeline)
* June
  - Nothing yet
* July
  - [Java Quarterly Patch Update](https://www.oracle.com/uk/security-alerts/)
* August
  - [Glibc](https://sourceware.org/glibc/wiki/Glibc%20Timeline)
  - [Go](https://go.dev/doc/devel/release)
* September
  - [Java](https://www.oracle.com/java/technologies/java-se-support-roadmap.html)
  - [LLVM](https://releases.llvm.org/)
* October
  - [Python](https://devguide.python.org/versions/)
  - [OpenSSL](https://openssl-library.org/post/2023-08-29-steps-forward/)
  - [Java Quarterly Patch Update](https://www.oracle.com/uk/security-alerts/)
* November
  - [PHP](https://php.watch/versions)
  - [.NET](https://dotnet.microsoft.com/en-us/)
* December
  - [Ruby](https://rubychangelog.com/timeline/) Merry Rubymas

End of EOL grace period:
* January
  - Nothing yet
* February
  - [Go](https://go.dev/doc/devel/release)
* March
  - Nothing yet
* April
  - [Python](https://devguide.python.org/versions/)
* May
  - Nothing yet
* June
  - [PHP](https://php.watch/versions)
* July
  - Nothing yet
* August
  - [Go](https://go.dev/doc/devel/release)
* September
  - [Ruby](https://rubychangelog.com/timeline/)
* October
  - [NodeJS](https://github.com/nodejs/Release/blob/main/schedule.json)
* November
  - Nothing yet
* December
  - Nothing yet

**Comment by pboling** [comment]
With regard to Ruby, and Rubygems, I received a question from a Wolfi contributor over here: https://gitlab.com/oauth-xx/version_gem/-/issues/3.  It took me a long time to respond, and now I fear the submitter doesn't use GitLab enough to see my response.  I need to know if building signed Rubygems without signing them is intentional.  It sounds very much like the antithesis of the whole project for a secure software supply chain, but I admit I don't know much about the specifics of how you build the distro.

More generally I'd love to understand if it is a good approach to allow a signed gem to be built without signing.  Is that a good idea, in general?  It doesn't seem hard to have a signing key such that signed gems could be signed by the distro when built, thus proving the provenance.

**Reply by kaniini** [reply]
Wolfi signs at the package (apk) level and provenance is proven through the apkdb.  This is pretty standard way for distributions to distribute things, including ruby gems.

I don’t know of a single distribution which bothers with signing content outside of the system package manager because it’s redundant: the gems included with distributions are only meant to be used in the context of that distribution’s package ecosystem.

**Reply by pboling** [reply]
Great!  I'll make it optional!

**Reply by xnox** [reply]
@pboling please start new discussions, this thread is meant for the calendar of regular upstream releases only; and not a general discussion. Thanks!

I will try to fork/move this into a separate topic, if I figure out how to do this.
